Add Deployments as Code Documentation (#1632)
astronomer/docs: astro/configure-deployment-resources.md

Worker and scheduler resources must be set for each Deployment and are managed separately from cluster-level infrastructure. Any additional components that Astro requires, including PgBouncer, KEDA, and the triggerer, are managed by Astronomer.

:::note

> You can manage a Deployemnent's resources with a Deployment file. Read [Deployments as Code](manage-deploymens-as-code.md) for more information.

Reviewer: ```suggestion
To manage Deployment resources with a Deployment file, see [Deployments as Code](manage-deploymens-as-code.md).
```
